pkg/waiter/waiter.go

```diff
@@ -0,0 +1,92 @@
+package waiter
+
+import (
+	"context"
+	"errors"
+)
+
+var (
+	// ErrWaiterClosed indicates the waiter has been closed prior to receiving a message.
+	ErrWaiterClosed = errors.New("waiter has been closed")
+)
+
+// Waiter waits for a value to be published to it, typically via another goroutine.
+//
+// The Manager that built the waiter is responsible for delivering the
```

Comment:
super nit: I guess we can call it caller instead of manager? Applies to the references in comments + the `managerDoneCh`.

Reply:
Interesting, I'm trying to think of a scenario in which a Waiter would exist with some other caller that isn't a Manager of sorts. The methods on the waiter aren't exactly called (except for Wait()), the key piece of functionality is that response chan that is managed outside the waiter.